rules/android_local_test/resources.bzl

```diff
@@ -0,0 +1,72 @@
+# Copyright 2020 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implementation."""
+
+load("//rules:acls.bzl", "acls")
+load(
+    "//rules:processing_pipeline.bzl",
+    "ProviderInfo",
+    "processing_pipeline",
+)
+load("//rules:resources.bzl", _resources = "resources")
+load("//rules:utils.bzl",  "get_android_toolchain", "utils")
+
+def _process_manifest(ctx, **unused_ctxs):
+    manifest_ctx = _resources.bump_min_sdk(
+        ctx,
+        manifest = ctx.file.manifest,
+        floor = _resources.DEPOT_MIN_SDK_FLOOR if (_is_test_binary(ctx) and acls.in_enforce_min_sdk_floor_rollout(str(ctx.label))) else 0,
+        enforce_min_sdk_floor_tool = get_android_toolchain(ctx).enforce_min_sdk_floor_tool.files_to_run,
+    )
+
+    return ProviderInfo(
+        name = "manifest_ctx",
+        value = manifest_ctx,
+    )
+
+def _process_resources_for_android_local_test(ctx, manifest_ctx, java_package, **unused_ctx):
+    packaged_resources_ctx = _resources.package(
+        ctx,
+        manifest = manifest_ctx.min_sdk_bumped_manifest,
+        manifest_values = utils.expand_make_vars(ctx, ctx.attr.manifest_values),
+        java_package = java_package,
+        use_legacy_manifest_merger = False,
+        should_throw_on_conflict = not acls.in_allow_resource_conflicts(str(ctx.label)),
+        deps = ctx.attr.deps + ctx.attr.associates,
+        aapt = get_android_toolchain(ctx).aapt2.files_to_run,
+        android_jar = ctx.attr._android_sdk[AndroidSdkInfo].android_jar,
+        busybox = get_android_toolchain(ctx).android_resources_busybox.files_to_run,
+        host_javabase = ctx.attr._host_javabase,
+    )
+    return ProviderInfo(
+        name = "packaged_resources_ctx",
+        value = packaged_resources_ctx,
+    )
+
+def _is_test_binary(ctx):
+    """Whether this android_binary target is a test binary.
+
+    Args:
+      ctx: The context.
+
+    Returns:
+      Boolean indicating whether the target is a test target.
+    """
+    return ctx.attr.testonly or ctx.attr.instruments or str(ctx.label).find("/javatests/") >= 0
+
+PROCESSORS_FOR_ANDROID_LOCAL_TEST = dict(
```

Comment:
Hi Mauricio,

Could you explain how this integrates into the rest of the pipeline? I think that the resource processor is supposed to read off this dict somewhere and then append the processors to the pipeline, but I don't see that change anywhere.

Reply:
we are using it in our implementation of kt_android_local_test in rules_kotlin https://github.com/mauriciogg/rules_kotlin/blob/97b008aefc006ab3b84b5022bb89c78236e0652b/kotlin/internal/jvm/android_impl.bzl#L201

https://github.com/mauriciogg/rules_kotlin/blob/97b008aefc006ab3b84b5022bb89c78236e0652b/kotlin/internal/jvm/android_resources.bzl#L77

though probably theres a better way to expose this, since this looks like an implementation detail

Reply:
I looked at this again today, and I think your implementation is slightly at odds with how resource processing for android_local_test in google3 works.

We are working on open-sourcing the android_local_test starlark code (concurrently with all the other stuff we're trying to do...). I can try to expedite the process for the resource processing parts specifically, if that would be helpful for you.

Reply:
android_local_test is now open sourced with resource processing.